### ISSUE TITLE
Previous search result and current search result are highlighted if they are next to one another

### ISSUE THREAD

**Issue by aaronjensen** [tmux/tmux#2603]
### Issue description

From `tmux -vv -Ltest -f/dev/null new`:

`echo eeeeeeeeeee`
`C-b [`
`C-r eee`

<img width="226" alt="CleanShot 2021-03-08 at 08 49 43@2x" src="https://user-images.githubusercontent.com/8588/110337180-3e0ccb00-7feb-11eb-9c68-b89ef2469789.png">

Both the preceding and current search results are highlighted when the cursor is at the beginning of one result and after another.

[tmux-client-6078.log](https://github.com/tmux/tmux/files/6102124/tmux-client-6078.log)
[tmux-out-6080.log](https://github.com/tmux/tmux/files/6102125/tmux-out-6080.log)
[tmux-server-6080.log](https://github.com/tmux/tmux/files/6102126/tmux-server-6080.log)

### Required information

```
Darwin i386
tmux next-3.3
tmux-256color
```

**Comment by topcat001** [comment]
This is an interesting bug! I'll take a look this evening and report back. Thanks you for reporting.

**Comment by topcat001** [comment]
Actually, this seems expected given the current code. If we search for `eee` then the cursor is positioned just after the match, and it highlights `eee`. However in that case the cursor is already on the next match and hence it is also highlighted. In other words, the highlight space is `eee.`. I don't see how we can consistently change this without also changing the conventions for positioning the cursor.

**Comment by aaronjensen** [comment]
@topcat001 yeah, I'm sure it has to do with https://github.com/tmux/tmux/commit/df7fbcd7a5a775586522380a70530721d1f2c151 being introduced, which is what introduced the two different (forward and reverse) conventions. 

It seems like it should only be highlighting the term based on the current search direction. That is, if the direction is forward it should highlight the term before. If the direction is backwards, it should highlight the term beginning with the cursor point.

**Comment by topcat001** [comment]
Yes this now works correctly.
